specs/DragStarting.md

```diff
@@ -0,0 +1,196 @@
+      {
+          if (m_dragOverrideMode != DragOverrideMode::OVERRIDE)
+          {
+              // If the event is marked handled, WebView2 will not execute its drag logic.
+              args->put_Handled(m_dragOverrideMode == DragOverrideMode::NOOP);
+              return S_OK;
+          }
+
+          COREWEBVIEW2_DRAG_EFFECTS allowedEffects;
+          POINT dragPosition;
+          wil::com_ptr<IDataObject> dragData;
+          wil::com_ptr<ICoreWebView2Deferral> deferral;
+
+          args->get_DragAllowedOperations(&allowedEffects);
+          args->get_DragPosition(&dragPosition);
+          args->get_DragData(&dragData);
+          args->GetDeferral(&deferral);
+
+          HRESULT hr = S_OK;
+          if (!m_oleInitialized)
+          {
+              hr = OleInitialize(nullptr);
+              if (SUCCEEDED(hr))
+              {
+                  m_oleInitialized = true;
+              }
+          }
+
+          if (!m_dropSource)
+          {
+              m_dropSource = Make<ScenarioDragDropOverrideDropSource>();
+          }
+
+          if (SUCCEEDED(hr))
+          {
+              DWORD effect;
+              DWORD okEffects = DROPEFFECT_NONE;
+              if (allowedEffects & COREWEBVIEW2_DRAG_EFFECTS_COPY)
+              {
+                  okEffects |= DROPEFFECT_COPY;
+              }
+              if (allowedEffects & COREWEBVIEW2_DRAG_EFFECTS_MOVE)
+              {
+                  okEffects |= DROPEFFECT_MOVE;
+              }
+              if (allowedEffects & COREWEBVIEW2_DRAG_EFFECTS_LINK)
+              {
+                  okEffects |= DROPEFFECT_LINK;
+              }
+
+              hr = DoDragDrop(dragData.get(), m_dropSource.get(), okEffects, &effect);
+          }
+
+          deferral->Complete();
+          args->put_Handled(TRUE);
+
+          return hr;
+      })
+      .Get(),
+  &m_dragStartingToken));
+```
+
+# API Details
+```C++
+/// Flags enum that represents the effects that a given WebView2 drag drop operation can have.
+[v1_enum]
+typedef enum COREWEBVIEW2_DRAG_EFFECTS {
+  /// Drag operation supports no effect.
+  COREWEBVIEW2_DRAG_EFFECTS_NONE = 0x0,
+  /// Drag operation supports copying data.
+  COREWEBVIEW2_DRAG_EFFECTS_COPY = 0x1,
+  /// Drag operation supports moving data.
+  COREWEBVIEW2_DRAG_EFFECTS_MOVE = 0x2,
+  /// Drag operation supports linking data.
+  COREWEBVIEW2_DRAG_EFFECTS_LINK = 0x4,
+} COREWEBVIEW2_DRAG_EFFECTS;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_DRAG_EFFECTS)")
+
+/// Event args for the `DragStarting` event.
+[uuid(edb6b243-334f-59d0-b3b3-de87dd401adc), object, pointer_default(unique)]
+interface ICoreWebView2StagingDragStartingEventArgs : IUnknown {
+  /// The operations this drag data supports.
+  [propget] HRESULT DragAllowedOperations([out, retval] COREWEBVIEW2_DRAG_EFFECTS* value);
+
+
+  /// The data being dragged.
+  [propget] HRESULT DragData([out, retval] IDataObject** value);
+
+  /// The position at which drag was detected.
```

Comment:
Please also say the coord space of the point - like what this is relative to and the units